parser/prism/Translator.cc

```diff
@@ -536,6 +536,26 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
                 return make_node_with_expr<parser::Integer>(move(sendNode), location, move(valueString));
             }
 
+            if (PM_NODE_FLAG_P(callNode, PM_CALL_NODE_FLAGS_ATTRIBUTE_WRITE)) { // like `self.foo = 1`
```

Comment:
Now that this code doesn't depend on the `args.front()`, I was able to move it up and out of the way before the argument translation is done.

This comes in handy in a later PR, which changes how args/blocks are processed below.